File: SOC/src/cc/game/soc/swing/MyToggleButton.java
package cc.game.soc.swing;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

import javax.swing.JCheckBox;

@SuppressWarnings("serial")
abstract class MyToggleButton extends JCheckBox implements ActionListener {

    MyToggleButton(String text, boolean on) {
        super(text, on);
        addActionListener(this);
    }
    
    @Override
    public void actionPerformed(ActionEvent e) {
        if (isSelected()) {
            onChecked();
        } else {
            onUnchecked();
        }
    }

    abstract void onChecked();
    abstract void onUnchecked();
}

File: SwingApps/apps/cc/applets/soc/MyToggleButton.kt
package cc.applets.soc

import java.awt.event.ActionEvent
import java.awt.event.ActionListener
import javax.swing.JCheckBox

internal abstract class MyToggleButton(text: String, on: Boolean) : JCheckBox(text, on), ActionListener {
	override fun actionPerformed(e: ActionEvent) {
		if (isSelected) {
			onChecked()
		} else {
			onUnchecked()
		}
	}

	abstract fun onChecked()
	abstract fun onUnchecked()

	init {
		addActionListener(this)
	}
}